zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/repo/HDFSNotebookRepo.java

```diff
@@ -0,0 +1,241 @@
+/**
+ * Copyright 2015 NAVER Corp. All rights Reserved.
+ */
+package org.apache.zeppelin.notebook.repo;
+
+import com.google.gson.Gson;
+import org.apache.commons.vfs2.FileObject;
+import org.apache.commons.vfs2.FileSystemManager;
+import org.apache.commons.vfs2.FileType;
+import org.apache.commons.vfs2.NameScope;
+import org.apache.commons.vfs2.VFS;
+import org.apache.zeppelin.conf.ZeppelinConfiguration;
+import org.apache.zeppelin.conf.ZeppelinConfiguration.ConfVars;
+import org.apache.zeppelin.file.HDFSCommand;
+import org.apache.zeppelin.file.HDFSCommand.Arg;
+import org.apache.zeppelin.file.HDFSFileInterpreter.AllFileStatus;
+import org.apache.zeppelin.file.HDFSFileInterpreter.OneFileStatus;
+import org.apache.zeppelin.notebook.Note;
+import org.apache.zeppelin.notebook.NoteInfo;
+import org.apache.zeppelin.user.AuthenticationInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.apache.zeppelin.file.HDFSFileInterpreter.HDFS_URL;
+import static org.apache.zeppelin.file.HDFSFileInterpreter.HDFS_MAXLENGTH;
+
+/**
+ *
+ * HDFSNotebookRepo : Using HDFS to backup and restore notebook
+ *
+ */
+public class HDFSNotebookRepo extends VFSNotebookRepo implements NotebookRepo {
+  Logger logger = LoggerFactory.getLogger(HDFSNotebookRepo.class);
+
+  public static final String HDFS_NOTEBOOK_DIR = "hdfs.notebook.dir";
+  static final String NOTE_JSON = "note.json";
+  static final String NOTE_JSON_TEMP = "_note.json";
+
+  ZeppelinConfiguration conf;
+  FileSystemManager fsManager;
+  boolean enableWebHDFS = true;
+  String hdfsUrl = null;
+  String hdfsUser = null;
+  int hdfsMaxLength = 0;
+  String hdfsNotebookDir = null;
+  HDFSCommand hdfsCmd = null;
+
+  public HDFSNotebookRepo(ZeppelinConfiguration conf) throws IOException {
+    super(conf);
+    this.conf = conf;
+    this.fsManager = VFS.getManager();
+    this.hdfsUrl = conf.getString(HDFS_URL, HDFS_URL, "http://localhost:50070/webhdfs/v1/");
+    this.hdfsMaxLength = conf.getInt(HDFS_URL, HDFS_MAXLENGTH, 100000);
+    this.hdfsNotebookDir = removeProtocol(conf.getString(HDFS_URL, HDFS_NOTEBOOK_DIR, "/tmp"));
+    this.hdfsUser = System.getenv("HADOOP_USER_NAME");
+    if (this.hdfsUser == null)
+      this.hdfsUser = System.getenv("LOGNAME");
+
+    if (this.hdfsUser == null) {
+      this.enableWebHDFS = false;
+    } else {
+      this.hdfsCmd = new HDFSCommand(hdfsUrl, hdfsUser, logger, this.hdfsMaxLength);
+      this.enableWebHDFS = checkWebHDFS();
+    }
+  }
+
+  public String removeProtocol(String hdfsUrl) {
+    String newUrl = hdfsUrl.replaceAll("/$", "");
+    if (newUrl.startsWith("hdfs://")) {
+      return "/" + newUrl.replaceAll("^hdfs://", "").split("/", 2)[1];
+    }
+    else
+      return newUrl;
```

Comment:
Please, make sure all the code follows [project style guide](https://zeppelin.apache.org/contribution/contributions.html#code-convention)